diff --git a/mingw-w64-qt5/PKGBUILD b/mingw-w64-qt5/PKGBUILD
index 1fc4da144..7d3ed086e 100644
--- a/mingw-w64-qt5/PKGBUILD
+++ b/mingw-w64-qt5/PKGBUILD
@@ -31,9 +31,9 @@ if [ "${_variant}" = "-static" ]; then
 #  _ltcg="yes"
   _ltcg="no"
 else
-  _namesuff=
-  _build_mode=("-release" "-force-debug-info" "-separate-debug-info")
-  _with_icu=yes
+  _namesuff=alst
+  _build_mode=("-release")
+  _with_icu=no
   _with_fontconfig=yes
   _with_openssl=yes
   _with_dbus=yes
@@ -50,7 +50,7 @@ else
   _build_examples=no
   _build_tools=yes
   _build_tests=no
-  _make_docs=yes
+  _make_docs=no
   _ltcg=no
 fi
 
@@ -124,7 +124,7 @@ license=('GPL3' 'LGPL' 'FDL' 'custom')
 if [ "${_variant}" = "-static" ]; then
   _qt5_prefix="${MINGW_PREFIX}/${_realname}"
 else
-  _qt5_prefix="${MINGW_PREFIX}"
+  _qt5_prefix="${MINGW_PREFIX}/opt/${_realname}"
 fi
 
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
@@ -162,7 +162,8 @@ optdepends=($([[ "$_variant" == "-shared" ]] && echo \
               "${MINGW_PACKAGE_PREFIX}-firebird2" \
               "${MINGW_PACKAGE_PREFIX}-postgresql")
             )
-makedepends=("bison"
+makedepends=("dos2unix"
+             "bison"
              "diffutils"
              "flex"
              "gperf"
@@ -226,7 +227,9 @@ source=(#https://download.qt.io/development_releases/qt/${pkgver%.*}/${_ver_base
         0125-qt5-windeployqt-fixes.patch
         0300-qt-5.8.0-cast-errors.patch
         0301-link-evr.patch
-        0302-ugly-hack-disable-qdoc-build.patch)
+        0302-ugly-hack-disable-qdoc-build.patch
+        qtbase.patch
+        qtdeclarative.patch)
 
 # Translates using cygpath according to the ${_make} being used
 # (so either mingw32-make or MSYS2 make can be used)
@@ -257,13 +260,16 @@ apply_patch_with_msg() {
   for _patch in "$@"
   do
     msg2 "Applying ${_patch}"
-    patch -Nbp1 -i "${srcdir}/${_patch}"
+    patch -sNbp1 -i "${srcdir}/${_patch}" || true
   done
 }
 
 prepare() {
+  set -x
+
   [[ -d ${srcdir}/${_pkgfqn} ]] && rm -rf ${srcdir}/${_pkgfqn}
-  tar -xJf ${srcdir}/${_pkgfqn}${_hotfix}.tar.xz -C ${srcdir} --exclude=${_pkgfqn}/{qtandroidextras,qtmacextras,qtwayland,qtwebengine,qtx11extras} || true
+
+  xz -dc "${srcdir}/${_pkgfqn}${_hotfix}.tar.xz" | tar -xf- -C ${srcdir} ${_pkgfqn}/{qtbase,qtquickcontrols,qtgraphicaleffects,gnuwin32,qtquickcontrols2,qtcharts,qtdeclarative,qttools,qttranslations,coin,qtwinextras,qt.pro,configure.bat,configure.json}
 
   cd ${srcdir}/${_pkgfqn}
 
@@ -326,6 +332,13 @@ prepare() {
     0300-qt-5.8.0-cast-errors.patch \
     0301-link-evr.patch
 
+  pushd qtbase
+    apply_patch_with_msg qtbase.patch
+  popd
+  pushd qtdeclarative
+    apply_patch_with_msg qtdeclarative.patch
+  popd
+
   # See: https://bugreports.qt.io/browse/QTBUG-37902
   # _ver_num=${_ver_base%%-*}
   # sed -i "s/^MODULE_VERSION = .*$/MODULE_VERSION = ${_ver_num}/g" qtenginio/.qmake.conf
@@ -428,6 +441,7 @@ build() {
   if [ "${_configure}" = "./configure" ]; then
     _extra_config+=("-no-glib")
   else
+    _extra_config+=("-no-glib")
     _extra_config+=("-make-tool")
     _extra_config+=("${_make}")
   fi
@@ -446,10 +460,13 @@ build() {
     _sql_config+=("-sql-sqlite")
     _sql_config+=("-qt-sqlite")
   else
-    _sql_config+=("-plugin-sql-ibase")
-    _sql_config+=("-plugin-sql-psql")
-    _sql_config+=("-plugin-sql-mysql")
-    _sql_config+=("-plugin-sql-odbc")
+    _sql_config+=("-no-sql-ibase")
+    _sql_config+=("-no-sql-mysql")
+    _sql_config+=("-no-sql-odbc")
+    _sql_config+=("-no-sql-psql")
+    _sql_config+=("-no-sql-sqlite2")
+    _sql_config+=("-sql-sqlite")
+    _sql_config+=("-qt-sqlite")
   fi
 
   #if [ "${_variant}" = "-static" ]; then
@@ -499,8 +516,6 @@ build() {
     fi
   else
     configure_opts+=("-shared")
-    configure_opts+=("-jasper")
-    configure_opts+=("-no-mng")
     if [ "$_with_dbus" == "yes" ]; then
       configure_opts+=("-dbus")
     else
@@ -530,16 +545,6 @@ build() {
   if [ "$_system_libpng" == "no" ]; then
     configure_opts+=("-qt-libpng")
   fi
-  if [ "$_system_libtiff" == "yes" ]; then
-    configure_opts+=("-system-tiff")
-  else
-    configure_opts+=("-qt-tiff")
-  fi
-  if [ "$_system_libwebp" == "yes" ]; then
-    configure_opts+=("-system-webp")
-  else
-    configure_opts+=("-qt-webp")
-  fi
   if [ "$_system_zlib" == "no" ]; then
     configure_opts+=("-qt-zlib")
   fi
@@ -551,11 +556,6 @@ build() {
   else
     configure_opts+=("-no-icu")
   fi
-  if [ "$_system_assimp" == "yes" ]; then
-    configure_opts+=("-system-assimp")
-  else
-    configure_opts+=("-qt-assimp")
-  fi
   if [ "$_system_doubleconversion" == "yes" ]; then
     configure_opts+=("-system-doubleconversion")
   else
@@ -577,6 +577,7 @@ build() {
   #fi
 
   ${_configure} \
+    -silent \
     -prefix ${QTDIR_WIN} \
     -datadir ${QTDIR_WIN}/share/qt5 \
     -archdatadir ${QTDIR_WIN}/share/qt5 \
@@ -591,7 +592,6 @@ build() {
     "${_extra_libpaths[@]}" \
     "${_extra_libs[@]}" \
     -no-iconv \
-    -no-gstreamer \
     "${_extra_config[@]}" \
     "${_sql_config[@]}" \
     "${configure_opts[@]}"
@@ -618,7 +618,8 @@ build() {
   # There is a race condition / dependency problem with libqtiff and dumpdoc.exe which
   # can happen at -j9 so make twice (dumpdoc.exe depends on libqtiff but can get built
   # first).
-  ${_make} ${MAKEFLAGS} || ${_make} ${MAKEFLAGS}
+  ${_make} ${MAKEFLAGS} &> /dev/null || perl -pi -e 's~\\(?![\r\n])~/~g' qttools/src/qdoc/Makefile.Release
+  ${_make} ${MAKEFLAGS} &> /dev/null || ${_make} ${MAKEFLAGS}
   # For quicker turn-around testing qtwebengine.
   # ${_make} module-qtbase
   # ${_make} module-qtwebengine
@@ -634,7 +635,7 @@ build() {
 
 check() {
   cd ${CARCH}
-  make check -j1 -k
+  make check -j1 -k &> /dev/null || make check -j1 -k
 }
 
 package_it() {
@@ -643,7 +644,9 @@ package_it() {
 
   export PATH=${pkgdir}${_qt5_prefix}/bin:${srcdir}/${CARCH}/qtbase/bin:${srcdir}/${CARCH}/qtbase/lib:${PATH}
 
-  ${_make} install
+  echo "Packaging..."
+  ${_make} install &> /dev/null || ${_make} install
+
   if [ "$_build_tools" == "yes" ]; then
     if [ "$_make_docs" == "yes" ]; then
       make docs
@@ -663,7 +666,7 @@ package_it() {
 
   # Remove dlls from lib/
   rm -f "${pkgdir}${_qt5_prefix}/lib"/*.dll
-  
+
   # Remove *.orig files
   find "${pkgdir}${_qt5_prefix}" -type f -name "*.orig" -exec rm -f {} \;
 
@@ -698,20 +701,20 @@ package_it() {
 
   find "${pkgdir}${_qt5_prefix}/lib/pkgconfig" -type f -name '*.pc' \
       -exec sed -i -e "s|${QT_PREFIX_WIN}|${_qt5_prefix}|g" {} \;
-      
+
   find "${pkgdir}${_qt5_prefix}/lib/pkgconfig" -type f -name '*.pc' \
       -exec sed -i -e "s|${PKGDIR_QT_PREFIX_WIN}|${_qt5_prefix}|g" {} \;
 
   if [ "${_variant}" = "-static" ]; then
     find "${pkgdir}${_qt5_prefix}/lib" -type f -name '*.cmake' \
       -exec sed -i -e "s|${MINGW_PREFIX_WIN}|${FAKE_PREFIX_FOR_REPLACE}/..|g" {} \;
-    
+
     find "${pkgdir}${_qt5_prefix}/share/qt5" -type f \( -name '*.pri' -o -name '*.prl' \) \
-      -exec sed -i -e "s|${MINGW_PREFIX_WIN}|${FAKE_PREFIX_FOR_REPLACE}/..|g" {} \;  
-    
+      -exec sed -i -e "s|${MINGW_PREFIX_WIN}|${FAKE_PREFIX_FOR_REPLACE}/..|g" {} \;
+
     find "${pkgdir}${_qt5_prefix}/lib" -type f \( -name '*.pri' -o -name '*.prl' \) \
       -exec sed -i -e "s|${MINGW_PREFIX_WIN}|${FAKE_PREFIX_FOR_REPLACE}/..|g" {} \;
-    
+
     find "${pkgdir}${_qt5_prefix}/lib/pkgconfig" -type f -name '*.pc' \
       -exec sed -i -e "s|${MINGW_PREFIX_WIN}|${MINGW_PREFIX}|g" {} \;
   fi
@@ -731,11 +734,11 @@ package_mingw-w64-i686-qt5-debug() {
   cp -rf ${srcdir}/${CARCH}-debug${MINGW_PREFIX} "${pkgdir}"/
 }
 
-package_mingw-w64-x86_64-qt5() {
+package_mingw-w64-x86_64-qt5alst() {
   package_it
 }
 
-package_mingw-w64-x86_64-qt5-debug() {
+package_mingw-w64-x86_64-qt5alst-debug() {
   cp -rf ${srcdir}/${CARCH}-debug${MINGW_PREFIX} "${pkgdir}"/
 }
 
diff --git a/mingw-w64-qt5/qt5-x86_64.install b/mingw-w64-qt5/qt5-x86_64.install
index 7cb6cf7a1..64d816cba 100644
--- a/mingw-w64-qt5/qt5-x86_64.install
+++ b/mingw-w64-qt5/qt5-x86_64.install
@@ -1,7 +1,7 @@
 post_install() {
-  if [ -f "mingw64/bin/qmake.exe" ]; then
-    cd mingw64/bin
-      ./qtbinpatcher --verbose --nobackup --force --old-dir=@@QT_REAL_PREFIX/dir@@
+  if [ -f "mingw64/opt/qt5alst/bin/qmake.exe" ]; then
+    cd mingw64/opt/qt5alst/bin
+      ./qtbinpatcher --verbose --nobackup --force
     cd -
   fi
 }
