diff --git a/src/network/access/qhttpnetworkreply.cpp b/src/network/access/qhttpnetworkreply.cpp
index 2024c0fefb..99714e8c2f 100644
--- a/src/network/access/qhttpnetworkreply.cpp
+++ b/src/network/access/qhttpnetworkreply.cpp
@@ -498,8 +498,12 @@ qint64 QHttpNetworkReplyPrivate::readStatus(QAbstractSocket *socket)
 
         // is this a valid reply?
         if (fragment.length() == 5 && !fragment.startsWith("HTTP/")) {
-            fragment.clear();
-            return -1;
+            if(fragment.startsWith("ICY ")) {
+                fragment.replace("ICY ", "HTTP/1.1 ");
+            } else {
+                fragment.clear();
+                return -1;
+            }
         }
     } while (haveRead == 1);
 
